torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1424,11 +1424,30 @@ at::Tensor XLANativeFunctions::index_add(const at::Tensor& self, int64_t dim,
 at::Tensor XLANativeFunctions::index_copy(const at::Tensor& self, int64_t dim,
                                           const at::Tensor& index,
                                           const at::Tensor& source) {
-  TORCH_LAZY_FN_COUNTER_TIMED_TRACING("xla::");
   XLATensorPtr self_tensor = bridge::GetXlaTensor(self);
-  return bridge::AtenFromXlaTensor(
-      tensor_methods::index_copy(self_tensor, dim, bridge::GetXlaTensor(index),
-                                 bridge::GetXlaTensor(source)));
+  absl::Span<const int64_t> self_tensor_dims =
+      self_tensor->shape().get().dimensions();
+
+  XLATensorPtr source_tensor = bridge::GetXlaTensor(source);
+  absl::Span<const int64_t> source_tensor_dims =
+      source_tensor->shape().get().dimensions();
+
+  XLATensorPtr index_tensor = bridge::GetXlaTensor(index);
+  absl::Span<const int64_t> index_tensor_dims =
```

Comment:
Maybe we can just query the length of the index tensor as it should be a vector.